ruma-events/src/room/message/content_serde.rs

```diff
@@ -1,24 +1,125 @@
-//! `Deserialize` implementation for MessageEventContent
+// //! `Deserialize` implementation for MessageEventContent
 
 use serde::{de, Deserialize};
 use serde_json::value::RawValue as RawJsonValue;
 
-use crate::{from_raw_json_value, room::message::MessageEventContent};
+use crate::{
+    from_raw_json_value,
+    room::message::{MessageEventContent, MessageType},
+};
+
+/// Helper struct to determine the msgtype, relates_to and new_content fields
+/// from a `serde_json::value::RawValue`
+#[derive(Debug, Deserialize)]
+struct MessageContentDeHelper {
+    msgtype: String,
+
+    #[serde(rename = "m.relates_to")]
+    relates_to: Option<Box<RawJsonValue>>,
+
+    #[cfg(feature = "unstable-pre-spec")]
+    #[serde(rename = "m.new_content")]
+    new_content: Option<Box<RawJsonValue>>,
```

Comment:
These should just be the field types from the `MessageEventContent` instead of `Box<RawJsonValue>`.